src/resolvers-cassandra/Messages/queries.js

```diff
@@ -119,46 +120,46 @@ function byLocation(args, res) { // eslint-disable-line no-unused-vars
 }
 
 /**
- * @param {site: string, originalSource: string, bbox: number[], mainTerm: string, filteredEdges: string[], langCode: string, limit: number, offset: number, fromDate: string, toDate: string, sourceFilter: string[], fulltextTerm: string} args
+ * @param {externalsourceid: string, bbox: number[], conjunctivetopics: string[], limit: number, pageState: number, fromDate: string, toDate: string, pipelinekeys: string[], fulltextTerm: string} args
  * @returns {Promise.<{runTime: string, type: string, bbox: number[], features: Feature[]}>}
- */
+ **/
 function byBbox(args, res) { // eslint-disable-line no-unused-vars
   return new Promise((resolve, reject) => {
     if (!args.bbox || args.bbox.length !== 4) return reject('Invalid bbox specified');
+    if(!args.conjunctivetopics || args.conjunctivetopics.length === 0) return reject('Empty conjunctive topic list specified');
 
-    const { fromDate, toDate } = parseFromToDate(args.fromDate, args.toDate);
     const [north, west, south, east] = args.bbox;
 
+    let tableName = 'eventplaces';
+    let tagsParams = [
+      ...fromTopicListToConjunctionTopics(args.conjunctivetopics),
+      args.toDate,
+      north,
+      east,
+      args.fromDate,
+      south,
+      west
+    ];
+
+    if(args.externalsourceid){
+      tagsParams.push(args.externalsourceid);
+      tableName = 'eventplacesbysource';
+    }
+
     const tagsQuery = `
     SELECT eventid
-    FROM fortis.eventplaces
+    FROM fortis.${tableName}
     WHERE conjunctiontopic1 = ?
     AND conjunctiontopic2 = ?
     AND conjunctiontopic3 = ?
-    AND pipelinekey = ?
-    AND externalsourceid = ?
-    AND (centroidlat, centroidlon, eventtime) <= (?, ?, ?)
-    AND (centroidlat, centroidlon, eventtime) >= (?, ?, ?)
-    LIMIT ?
+    AND (eventtime, centroidlat, centroidlon) <= (?, ?, ?)
+    AND (eventtime, centroidlat, centroidlon) >= (?, ?, ?)
+    ${args.externalsourceid ? ' AND externalsourceid = ?' : ''}
+    ${args.pipelinekeys ? ` AND pipelinekey IN('${args.pipelinekeys.join('\',\'')}')` : ''}
```

Comment:
It would be safer to insert placeholders (`?`) here and add the actual pipelinekey values via the params array.